### PR TITLE
Feat: Endpoints for Fetch Product Stock

### PR DESCRIPTION
**Pull request by Neche-Stephen**
### How should this be manually tested?
---

Access the GET /api/v1/products/:productId/stock endpoint with a valid productId.
• Expected response are the required fields in JSON format. 
```json
{
  "productId": "12345",
  "currentStock": 100,
  "lastUpdated": "2024-07-18T10:00:00Z"
}
```
Access the GET /api/v1/products/:productId/stock endpoint with an invalid productId.
• Expected response is an appropriate error message and status code.
```json
{
"statusCode": 400,
"message": "Bad Request"
}
```

### Checklist of what you did
---

- [x] Implemented a functional GET /api/v1/products/:productId/stock endpoint.
- [x] Ensured that the endpoints return the required fields in JSON format.
- [x] Ensured proper error handling mechanisms are in place.

### Reference Issue
---

[FEAT]: Endpoint for Fetch Product Stock ([#284](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/284))

### CODE REVIEW

File: Contributors.md
Comment (Fifetoyi) [inline Contributors.md:1]: not needed

File: pom.xml
Comment (Fifetoyi) [inline pom.xml:1]: there is already a flyway plugin in the codebase

File: Contributors.md
Comment (Neche-Stephen) [reply Contributors.md:1]: Removed

File: pom.xml
Comment (Neche-Stephen) [reply pom.xml:1]: I have removed mine

File: pom.xml [hunk 1]
```diff
@@ -164,7 +164,22 @@
...
 				<artifactId>spring-boot-maven-plugin</artifactId>
 			</plugin>
+
+			<plugin>
+				<groupId>org.flywaydb</groupId>
+				<artifactId>flyway-maven-plugin</artifactId>
+				<version>10.10.0</version>
+				<configuration>
+					<url>jdbc:postgresql://localhost:5432/HNG</url>
+					<user>postgres</user>
```
[the remark this answers is not in the document]
Comment (Neche-Stephen) [reply pom.xml]: Removed as well.

File: src/main/resources/db/migration/V7__alter_products_table_schema.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+--Added needed columns
+ALTER TABLE products
```
Comment (lurldgbodex) [inline src/main/resources/db/migration/V7__alter_products_table_schema.sql:2]: There is a version 7 of the migration file. update your version

**Review by lurldgbodex — CHANGES_REQUESTED** [review]
update your migration version

**Pull request closed without merging**